packages/cozy-client/src/store/queries.spec.js

```diff
@@ -256,4 +256,40 @@ describe('makeSorterFromDefinition', () => {
     const sorted = sorter(files)
     expect(sorted.map(x => x._id)).toEqual(['4', '5', '3', '2', '1'])
   })
+
+  it('should handle objects with deep properties', () => {
+    const q = Q('io.cozy.files').sortBy([{ 'name.label': 'asc' }])
+    const sorter = makeSorterFromDefinition(q)
+    const files = [
+      { _id: '1', name: { label: 'C' } },
+      { _id: '2', name: { label: 'B' } },
+      { _id: '3', name: { label: 'a' } },
+      { _id: '4', name: { label: 'A' } },
+      { _id: '5', name: { label: 'b' } },
+      { _id: '6', name: { label: [] } },
+      { _id: '7', name: { label: '' } },
+      { _id: '8', name: { label: 1 } },
+      { _id: '9', name: { label: {} } },
+      { _id: '10', name: { label: false } },
+      { _id: '11', name: { label: null } },
+      { _id: '12', name: { label: true } },
+      { _id: '13', name: { label: undefined } }
```

Comment:
Why do we have all those cases here ? Aren't we testing the behavior of Array.sort from _id 6 to _id 13 ?

Reply:
Yes, that's exactly right.

Reply:
I'm will delete that to stay within an application perimeter.